Mathlib/RingTheory/NonZeroDivisors.lean

```diff
@@ -37,6 +35,19 @@ def nonZeroDivisors (R : Type*) [MonoidWithZero R] : Submonoid R where
 /-- The notation for the submonoid of non-zerodivisors. -/
 scoped[nonZeroDivisors] notation:9000 R "⁰" => nonZeroDivisors R
 
+/-- Let `R` be a monoid with zero and `M` an additive monoid with an `R`-action, then the collection
+of non-zero smul-divisors forms a submonoid. These elements are also called `M`-regular.-/
+def nonZeroSMulDivisors (R : Type*) [MonoidWithZero R] (M : Type _) [AddMonoid M] [MulAction R M] :
```

Comment:
Does this need `AddMonoid M`?

Reply:
Indeed, it is sufficient to assume `[Zero M]` only